core-bundle/src/Entity/TrustedDevice.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="Contao\CoreBundle\Repository\TrustedDeviceRepository")
+ * @ORM\Table(name="tl_trusted_device")
+ */
+class TrustedDevice
+{
+    /**
+     * @var int
+     *
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    protected $id;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    protected $created;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(type="string", nullable=true)
+     */
+    protected $userClass;
+
+    /**
+     * @var int|null
+     *
+     * @ORM\Column(type="integer", nullable=true)
+     */
+    protected $userId;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(type="text", name="cookie_value")
+     */
+    protected $cookieValue;
+
+    /**
+     * @var int
+     *
+     * @ORM\Column(type="integer")
+     */
+    protected $version;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(type="text", name="user_agent", nullable=true)
+     */
+    protected $userAgent;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(type="string", name="ua_family", nullable=true)
+     */
+    protected $uaFamily;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(type="string", name="os_family", nullable=true)
+     */
+    protected $osFamily;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(type="string", name="device_family", nullable=true)
+     */
+    protected $deviceFamily;
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function setId(int $id): self
+    {
+        $this->id = $id;
+
+        return $this;
+    }
+
+    public function getCreated(): \DateTime
+    {
+        return $this->created;
+    }
+
+    public function setCreated(\DateTime $created): self
+    {
+        $this->created = $created;
+
+        return $this;
+    }
+
+    public function getUserClass(): string
+    {
+        return $this->userClass;
+    }
+
+    public function setUserClass(?string $userClass): self
```

Comment:
How can the user class ever be empty/null?

Reply:
Afair, Doctrine initializes every object with null values.

Reply:
the DB schema is not related to the setter method (methods are not used by Doctrine). But as you cannot really have a TrustedDevice without user, I would suggest to add a constructor with User argument and set these parameters automatically (and remove the setters, as you can never change them).

Reply:
> methods are not used by Doctrine

They are, while retrieving from the database via repository.
I'll check and remove the unused ones.

Reply:
Fixed in 82e6f10.